### ISSUE TITLE
[A11ySev2] [Screen reader-Dashboards with Grafana - New Dashboard- Add another transformation]: Screen reader does not announce information after entering text in the search field under the Add another Transformation tab.

### ISSUE THREAD

**Issue by LinglingTong** [grafana/grafana#118775]
## Steps to reproduce

Repro Steps: 
  
1. Open 'Azure Monitor dashboards with Grafana' in Azure Portal
2. "Monitor | Dashboards with Grafana | Gallery" screen appears with already Dashboards with Grafana tab selected.  
3. Navigate to the all the controls available on Dashboards with Grafana blade like tool bar , controls under All Dashboards tab. 
4. Navigate New menu present in the tool bar and select New Dashboard.  
5. Navigate to "Add visualization" button and hit ENTER key. "Select data source" popup will appear. 
6. Navigate to the TestData list item select it. 
7. Navigate to the Transformation tab and select add transformation button. 
8. Navigate to all controls under Add another transformation dialog.  
9. Observe that the screen reader does not provide feedback or announce any information about the input or search results. 

**Actual Result:**
The screen reader remains silent after entering text in the search field, providing no feedback about the action or any dynamically updated information.  
**Observation**:  
The lack of feedback may be due to the absence of an aria-live region or a similar mechanism for dynamically announcing status messages. Implementing this feature will ensure compliance with accessibility standards and improve the user experience for screen reader users.  
**Issue1**: Same issue repro for the search edit field under the 'Select a resource' tab.  
**Navigation**: Open https://portal.azure.com/#view/Microsoft_Azure_Monitoring/AzureMonitoringBrowseBlade/~/AzureGrafanaReactView  > Select New button button > Invoke New dashboard control> Select “ Add visualization” button > Select testdata list item> Select Back to dashboard in the tool bar > Select Add visualization from the add menu button in the tool bar > Invoke ‘Select a resource’ button under Queries tab > Observe that the screen reader does not provide feedback or announce any information about the input or search results.  
**Issue2**: Same state issue repro for the "copy to clipboard" button control after activating the copy to clipboard  control under the query tab.   
**Navigation**: Open https://portal.azure.com/#view/Microsoft_Azure_Monitoring/AzureMonitoringBrowseBlade/~/AzureGraf…  > Select New button > Invoke New dashboard control> Select “ Add visualization” button > Select testdata list item> Select Last 7 days in the combo box above new panel> Activate Query inspector in the Queries tab > Activate refresh button in the query tab> Invoke copy to clipboard button > Observe that the screen reader does not announce "Copied." After activating copy to clipboard button.  

**Expected Result:**
The screen reader should announce relevant feedback after text is entered in the search field. For example:  

- If results are dynamically updated, the screen reader should announce the number of matching results (e.g., "5 results found"). 
- If no results are found, the screen reader should announce "No results found."

**Relevant WCAG Criteria:** [4.1.3 Status Messages](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html) (Level AA)

## Environment

- **Grafana version:** 12.2.0
- **Data source type & version:**
- **User OS & Browser:** Version 25H2 (OS Build 26200.7093) / Chrome Version 142.0.7444.177 (Official Build) (64-bit)
- **Others:**

**Comment by LinglingTong** [comment]
<img width="1919" height="1085" alt="Image" src="https://github.com/user-attachments/assets/bb528659-d9f9-4f9e-b1df-c00185c7cd88" />
<img width="1919" height="1147" alt="Image" src="https://github.com/user-attachments/assets/f83031f6-ef31-4b44-81c0-8982a724b946" />
<img width="1915" height="1070" alt="Image" src="https://github.com/user-attachments/assets/49b8951c-e180-4ef9-90fb-e0ba1469acdf" />

**Comment by joshhunt** [comment]
There's two issues here:
 - neither the [transformations drawer](https://github.com/grafana/grafana/blob/1e1404ad2989da87912bc3645f9ccba67fe04f14/public/app/features/dashboard/components/TransformationsEditor/TransformationPickerNg.tsx#L65), or the [transformations list](https://github.com/grafana/grafana/blob/1e1404ad2989da87912bc3645f9ccba67fe04f14/public/app/features/dashboard-scene/panel-edit/PanelEditNext/QueryEditor/Body/TransformationTypePicker.tsx#L71) when `queryEditorNext` is enabled uses an aria-live region to announce the search results
 - the Copy to Clipboard button doesn't announce 'copied!' from the inline toast when clicked
    - https://github.com/grafana/grafana/issues/128685

**Comment by joshhunt** [comment]
Copy to clipboard button should be fixed in https://github.com/grafana/grafana/issues/118775

**Comment by v-kavyagupta** [comment]
We verified this issue on the URL: https://ms.portal.azure.com/?feature.heliosaksregion=eus2e&grafanaVersion=13.2.0#view/Microsoft_Azure_Monitoring/AzureMonitoringBrowseBlade/~/AzureGrafanaReactView
Issue 1 is still repro; Screen reader does not announce information after entering text in the search field under Select a resource search field.

<img width="1904" height="927" alt="Image" src="https://github.com/user-attachments/assets/f65ef034-29dc-4731-8342-55a9f49e30e6" />
